### PR TITLE
neptune.ai integration

### PR DESCRIPTION
**Pull request by AleksanderWWW**
## Before submitting

- [x] Was this discussed/approved via a GitHub issue? (no need for typos and docs improvements)
- [x] Did you create/update your **configuration file**?
- [x] Did you **set `runtimes` in config** for GitHub action integration?
- [x] Did you **add your config to CI** in Azure pipeline (only projects with 100+ GitHub stars)?
- [x] Are all integration **tests passing**?

## What does this PR do? \[optional\]

it adds neptune.ai integration for lightning to the ecosystem

Fixes # ([807](https://github.com/neptune-ai/neptune-client/issues/807))

OR link your project for clarity.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: configs/neptune-ai/lightning_integration.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+target_repository:
+  HTTPS: https://github.com/neptune-ai/neptune-client
+  # OPTIONAL, for some protected repository
```
Comment (Raalsky) [inline configs/neptune-ai/lightning_integration.yaml:3]: Let's just drop all these non-used parameters with comments related (`username` etc.)

File: configs/neptune-ai/lightning_integration.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  email: [aleksander.wojnarowicz@neptune.ai, backend@neptune.ai]
+
+# OPTIONAL, running before installing your project
+# WARNING: this is custom for given OS, we do not manage any inter OS compatibility
+before_install:
+  - pip install Cython==3.56
+
+# OPTIONAL, if any installation require some env. variables
+env:
+  HOROVOD_BUILD_ARCH_FLAGS: "-mfma"
```
Comment (Raalsky) [inline configs/neptune-ai/lightning_integration.yaml:31]: Are these `horovod` reletead envs actually relevant?

File: configs/neptune-ai/lightning_integration.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+contact:
+  slack:
+    - Aleksander Wojnarowicz
+  email: [aleksander.wojnarowicz@neptune.ai, backend@neptune.ai]
+
+# OPTIONAL, running before installing your project
+# WARNING: this is custom for given OS, we do not manage any inter OS compatibility
+before_install:
+  - pip install Cython==3.56
```
Comment (Raalsky) [inline configs/neptune-ai/lightning_integration.yaml:27]: Is this relevant and required?

File: configs/neptune-ai/lightning_integration.yaml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    checkout: 0.29.24
+
+# OPTIONAL, running before installing your project
+# WARNING: this is custom for given OS, we do not manage any inter OS compatibility
+# before_test:
+#  - wget https://pl-flash-data.s3.amazonaws.com/titanic.zip
+
+testing:
+  dirs:
+    - .
```
Comment (Raalsky) [inline configs/neptune-ai/lightning_integration.yaml:52]: Probably should be `e2e_tests`

File: configs/neptune-ai/lightning_integration.yaml [hunk 5]
```diff
@@ -0,0 +1,42 @@
+target_repository:
+  HTTPS: https://github.com/neptune-ai/neptune-client
+  # OPTIONAL, checkout a particular branch or a tag
+  checkout: aw/adjust_e2e_tests
+  # copy some  test from the target repository
+  copy_tests:
+    - e2e_tests/integrations/__init__.py
+    - e2e_tests/integrations/test_pytorch_lightning.py
```
Comment (ethanwharris) [inline configs/neptune-ai/lightning_integration.yaml:8]:
```suggestion
    - e2e_tests/integrations/test_lightning.py
```

**Review by ethanwharris — COMMENTED** [review]
Small bug 😃

**Review by Raalsky — APPROVED** [review]

**Review by ethanwharris — APPROVED** [review]
Awesome, thanks for this @AleksanderWWW LGTM :smiley:

File: configs/neptune-ai/lightning_integration.yaml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    - e2e_tests/integrations/test_lightning.py
+    - e2e_tests/conftest.py
+    - e2e_tests/base.py
+    - e2e_tests/utils.py
+    - e2e_tests/exceptions.py
+    - e2e_tests/pytest.ini
+
+contact:
+  slack:
+    - Aleksander Wojnarowicz
```
Comment (Borda) [inline configs/neptune-ai/lightning_integration.yaml:17]:
```suggestion
    # Aleksander Wojnarowicz
    - U046VPPNVE0
```

File: configs/neptune-ai/lightning_integration.yaml [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+# OPTIONAL, if any installation require some env. variables
+env:
+  NEPTUNE_API_TOKEN: ANONYMOUS
+  NEPTUNE_LIGHTNING_ECOSYSTEM_CI_PROJECT: common/lightning-ecosystem-ci
+
+dependencies:
+  - name: pytorch-lightning
+    HTTPS: https://github.com/Lightning-AI/lightning.git
+    checkout: release/pytorch
+
```
Comment (Borda) [inline configs/neptune-ai/lightning_integration.yaml:33]:
```suggestion
  - name: Faker
    checkout: 15.0.0

```

File: configs/neptune-ai/lightning_integration.yaml [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    - e2e_tests/exceptions.py
+    - e2e_tests/pytest.ini
+
+contact:
+  slack:
+    - Aleksander Wojnarowicz
+  email: [aleksander.wojnarowicz@neptune.ai, backend@neptune.ai]
+
+before_install:
+  - pip install Faker==15.0.0
```
Comment (Borda) [inline configs/neptune-ai/lightning_integration.yaml:22]:
```suggestion
```
this could be part of denedencies

**Review by Borda — APPROVED** [review]

**Pull request merged**